CanDIG/CanDIGv2
# Integrating OPA into the stack

Set up the stack as described in docs/install-docker.md.

To test:

* Get a user token, where the values for the data parameters are found in the files in `tmp/secrets`:
```
curl -X "POST" "http://auth.docker.localhost:8080/auth/realms/candig/protocol/openid-connect/token" \
     -H 'Content-Type: application/x-www-form-urlencoded; charset=utf-8' \
     --data-urlencode "client_id=local_candig" \
     --data-urlencode "client_secret=<value in keycloak-client-local_candig-secret>" \
     --data-urlencode "grant_type=password" \
     --data-urlencode "username=<value in keycloak-test-user>" \
     --data-urlencode "password=<keycloak-test-user-password>" \
     --data-urlencode "scope=openid"
```
* Using the access_token from that, get the results from:
```
curl "http://docker.localhost:5080/katsu/api/phenopackets" \
     -H 'Authorization: Bearer <access_token>'
```
You should have 4 datasets, including registered3 and controlled4.

If you follow the same steps but with the user/pass for keycloak-test-user-2, you should get 3 datasets, including controlled5 (but not registered3 or controlled4).
